content/2017.04.getting-started-with-elm.rst

```diff
@@ -0,0 +1,433 @@
+Getting started with Elm
+########################
+
+:lang: en
+:date: 2017-04-13
+:url: getting-started-with-elm
+:summary: What is Elm? how does it work? How can I try it really fast?
+
+Introduction
+============
+
+The past two weeks we have been traveling and meeting a bunch of
+developers.
+
+It was a chance for us to tell them about elm through talks at
+conference and to train a little and short one hour workshop.
+
+You can find the slides of the two talks here:
+
+- `a short introduction to Elm by Niko <http://slides.com/n1k0/elm>`_
+- `Talking JSON with Elm by Natim <http://natim.github.io/django-slides/elm-json/>`_
+
+However, this article is a summary of the take aways of this workshop
+for thoose who would like to know a bit more about Elm.
+
+
+What is Elm and why should I be concerned?
+==========================================
+
+Elm is a toolkit that is focused on making sure your app state and the
+DOM (html nodes) that reflects it are always in sync and that updates
+on your app state are reflected on what the user is seeing.
+
+If you have been doing some frontend during the last couple of years,
+you might have seen that the JavaScript ecosystem is a bit wild.
+
+Keeping up-to-date with the growing number of competing frameworks can
+take a lot of energy, not to mention trying your app on all the
+available browser version to make sure it works well everywhere.
+
+Training a team to build an app with React.js, Angular2.js, Vue.js or
+Ember.js as well as looking for the right boilerplate to get started
+with your project are good examples of how hard the learning curve can
+be.
+
+In the other hands, writing tests for the front-end can be tidious and
+supporting Flow or TypeScript proved to be yet another layer to handle
+in your stack.
+
+If you are already up-to-speed with all these stacks, then you should
+probably keep using them.
+
+However if you don't know which one to choose and that you are looking
+for a painless way to build frontend application, read on.
+
+
+How is Elm different?
+=====================
+
+First of all, using Elm, loading your app is all the Javascript you
+will write. Your application will be written in Elm and compiled into
+Javascript.
+
+All the mistake you will do will be handled at compilation, which
+means that you will not have error at the javascript level on runtime.
+
+Because you will have a compilation step, Elm will force you to handle
+every possible cases of your model state to be handled on the display
+side.
+
+The tooling is great, elm-live will reflect your code update in
+realtime in the browser, you have a debugger that will show you the
+list of events and show you the state of your application at a given
+time. You can even go back in time and replay events. elm-format will
+also format your code automatically to avoid fighting with coding
+styles.
```

Comment:
Commas are not what you want here. I think this is meant to be a loose list, in which case I'd suggest periods after `great` and `browser`. I don't like the wording of `reflect your code update in realtime in the browser` but I can't think of anything better off the top of my head. `fighting with coding styles` makes it sound like the coding style is doing the fighting; I'd suggest `fighting over coding styles`.